File: react-java/backend/src/main/java/com/clt/hrm/core/administration/role/dtos/SearchRoleDto.java
package com.clt.hrm.core.administration.role.dtos;

import com.clt.hrm.infra.common.dtos.SearchBaseDto;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class SearchRoleDto extends SearchBaseDto {
	private String roleId;
    private String roleCd;
    private String roleNm;
    private String pgmId;
    private String permId;
}

File: nestjs/backend/src/apps/administration/roles/dto/search-role.dto.ts
import { IsOptional, IsString } from 'class-validator';
import { SearchBaseDto } from '@infra/common/dtos';

export class SearchRoleDto extends SearchBaseDto {
  @IsOptional()
  @IsString()
  roleId?: string;

  @IsOptional()
  @IsString()
  roleCd?: string;

  @IsOptional()
  @IsString()
  roleNm?: string;

  @IsOptional()
  @IsString()
  pgmId?: string;

  @IsOptional()
  @IsString()
  permId?: string;
}
